### PR TITLE
WIP HATCH-512 Add alwaysSorted prop and initialSort state

### PR DESCRIPTION
**Pull request by aminnoah**
# Description

# Jira ticket links

# Test Plan

# Definition of done

- [x] Does new code have 90% testing coverage (100% for `core`) of both unit and E2E tests?
- [x] Is code secure? If applicable, add security notes to the description.
- [ ] Do all new TODO comments have Jira links with enough information?
- [ ] Has the browser error-console been reviewed to not contain new errors introduced by these code changes?
- [ ] The site looks “good” above 1000px width.
- [ ] The site looks “good” when the window height is small. No double scrollbars.
- [ ] Were the changes tested to ensure 508 compliance?

### CODE REVIEW

File: packages/react-ui/src/hooks/useSort.test.ts [hunk 1]
```diff
@@ -63,9 +68,84 @@ describe("useSort", () => {
...
+    const { result } = renderHook(() =>
+      useSort({ direction: undefined, sortBy: undefined, alwaysSorted: true }),
+    )
+
+    // initial state
+    expect(result.current.sort).toEqual({
+      direction: undefined,
+      sortBy: undefined,
+      alwaysSorted: true,
+    })
```
Comment (arthurpankiewicz) [inline packages/react-ui/src/hooks/useSort.test.ts:88]: is this valid? if `alwaysSorted` is true shouldn't `sortBy` and `direction` be defined? we can enforce this with typescript as well but just wondering if there's a case I'm missing/unaware of (not exactly sure on ticket criteria 😅)

File: packages/react-ui/src/hooks/useSort.ts [hunk 2]
```diff
@@ -18,13 +19,15 @@ export default function useSort(
...
+    if (alwaysSorted && sort.direction === "desc") {
+      setSort({ sortBy, direction: "asc", alwaysSorted })
+    } else if (sort.sortBy === undefined || sort.sortBy !== sortBy) {
+      setSort({ sortBy, direction: "asc", alwaysSorted })
     } else if (sort.direction === "asc") {
-      setSort({ sortBy, direction: "desc" })
+      setSort({ sortBy, direction: "desc", alwaysSorted })
     } else {
-      setSort({ sortBy: undefined, direction: undefined })
+      setSort({ sortBy: undefined, direction: undefined, alwaysSorted })
```
Comment (arthurpankiewicz) [inline packages/react-ui/src/hooks/useSort.ts:30]: similarly here, if `alwaysSorted` is passed in as `true`, should the other arguments be defined?

File: packages/react-ui/src/hooks/useSort.test.ts (hunk 1 above)
Comment (aminnoah) [reply packages/react-ui/src/hooks/useSort.test.ts:88]: `alwaysSorted` refers to the action of the sort after the third click when the `direction` equals `desc`. `alwaysSorted' changes the default behavior of the third click from setting `direction`  and `sortBy` to `undefined` to restart the sort from `sortBy` and `direction` set to `asc`.

Hopefully that makes sense 😅

File: packages/react-ui/src/hooks/useSort.ts (hunk 2 above)
Comment (aminnoah) [reply packages/react-ui/src/hooks/useSort.ts:30]: No if `alwaysSorted` is true, it won't take action until the `direction` is being changed from `desc`

File: packages/react-ui/src/hooks/useSort.test.ts (hunk 1 above)
Comment (arthurpankiewicz) [reply packages/react-ui/src/hooks/useSort.test.ts:88]: got it, makes sense!

File: packages/react-ui/src/hooks/useSort.ts (hunk 2 above)
Comment (arthurpankiewicz) [reply packages/react-ui/src/hooks/useSort.ts:30]: 👍

**Pull request merged**